internal/viridian/cluster.go

```diff
@@ -0,0 +1,120 @@
+package viridian
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/rand"
+	"os"
+	"path"
+	"time"
+)
+
+type createClusterRequest struct {
+	KubernetesClusterID int         `json:"kubernetesClusterId"`
+	Name                string      `json:"name"`
+	ClusterTypeID       ClusterType `json:"clusterTypeId"`
+	PlanName            ClusterPlan `json:"planName"`
+}
+
+type createClusterResponse Cluster
+
+func (a API) CreateCluster(ctx context.Context, name string, plan ClusterPlan, k8sClusterID int, isDev bool) (Cluster, error) {
+	if name == "" {
+		name = clusterName()
+		if name == "" {
+			return Cluster{}, errors.New("could not generate cluster name")
+		}
+	}
+	if plan == "" {
+		plan = ClusterPlanServerless
+	}
+	ct := ClusterTypeProd
+	if isDev {
+		ct = ClusterTypeDev
+	}
+	c := createClusterRequest{
+		KubernetesClusterID: k8sClusterID,
+		Name:                name,
+		ClusterTypeID:       ct,
+		PlanName:            plan,
+	}
+	cluster, err := doPost[createClusterRequest, createClusterResponse](ctx, "/cluster", a.Token(), c)
+	if err != nil {
+		return Cluster{}, fmt.Errorf("creating cluster: %w", err)
+	}
+	return Cluster(cluster), nil
+}
+
+func clusterName() string {
+	cwd, err := os.Getwd()
+	if err != nil {
+		return ""
```

Comment:
Could you return a default string if there's an error getting cwd, like: `cluster` ?